src/app-layer-htp.c

```diff
@@ -1703,9 +1703,8 @@ static int HTPCallbackRequestStart(const htp_connp_t *connp, htp_tx_t *tx)
         }
         tx_ud->tx_data.file_tx = STREAM_TOSERVER | STREAM_TOCLIENT; // each http tx may xfer files
         htp_tx_set_user_data(tx, tx_ud);
-    } else {
```

Comment:
Funny how this error came from the pattern only use tx_ud if it is not NULL but here we allocate it if it is NULL (and check alloc return)